### PR TITLE
Calling C functions from BigQuery

### PR DESCRIPTION
**Pull request by campoy**

### CODE REVIEW

File: content/post/calling-c-functions-from-bigquery.md [hunk 1]
```diff
@@ -0,0 +1,372 @@
+---
+author: campoy
+date: 2018-03-06
+title: "Calling C functions from BigQuery with Web Assembly"
+draft: false
+image: /post/c-on-bigquery/banner.png
+description: "TODO"
```
Comment (dpordomingo) [inline content/post/calling-c-functions-from-bigquery.md]: leftover

**Comment by dpordomingo** [comment]
I saw you created a lot of files under [static/post/c-on-bigquery](https://github.com/src-d/blog/tree/ee568cd10eead433afe42a4e3d3a1d0e64296eeb/static/post/c-on-bigquery) but they're not used nor linked in the post.


It would be great if we could have a shortcode to reuse and render its content (avoiding duplicated code), something like this:

```markdown
Let's now write a `main.js` that loads the `wasm` file and uses its `sum` function.
 
{{% code "javascript" "c-on-bigquery/main.js" %}}

The code might seem complex, but if you ignore all the code before the call [...]
```
that would render:

sortcode-code

File: content/post/calling-c-functions-from-bigquery.md [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+categories: ["culture", "why-i-joined"]
```
[the remark this answers is not in the document]
Comment (campoy) [reply content/post/calling-c-functions-from-bigquery.md]: yes!

File: content/post/calling-c-functions-from-bigquery.md (hunk 1 above)
Comment (campoy) [reply content/post/calling-c-functions-from-bigquery.md]: done

File: content/post/calling-c-functions-from-bigquery.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (campoy) [reply content/post/calling-c-functions-from-bigquery.md]: done

**Comment by campoy** [comment]
This is now ready for reviews! Depends on https://github.com/src-d/blog/pull/198 being merged

**Comment by campoy** [comment]
Planned to be released tomorrow at 8am PST.

File: content/post/calling-c-functions-from-bigquery.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">OMG I just ran a function written in C on <a href="https://twitter.com/hashtag/BigQuery?src=hash&amp;ref_src=twsrc%5Etfw">#BigQuery</a> by using <a href="https://twitter.com/hashtag/webassembly?src=hash&amp;ref_src=twsrc%5Etfw">#webassembly</a> 🔥🔥🔥<br><br>The only problem? It takes 3 seconds to simply double a number 😩<br>cc <a href="https://twitter.com/felipehoffa?ref_src=twsrc%5Etfw">@felipehoffa</a><br><br>gist: <a href="https://t.co/O8IaNrEggj">https://t.co/O8IaNrEggj</a> <a href="https://t.co/gB0bH5tfII">pic.twitter.com/gB0bH5tfII</a></p>&mdash; Francesc (@francesc) <a href="https://twitter.com/francesc/status/951935862467514368?ref_src=twsrc%5Etfw">January 12, 2018</a></blockquote>
+<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+### Making it go faster
+
+BigQuery is not necessarily fast when you take into account its latency, but it's incredibly
+fast when you measure its throughput!
+
+Taking that into account [Felipe Hoffa](), Developer Advocate for Google Cloud Platform and
```
Comment (bzz) [inline content/post/calling-c-functions-from-bigquery.md]: https://twitter.com/felipehoffa

File: content/post/calling-c-functions-from-bigquery.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+What if you delete `sum.so` and run `dynamic`?
+
+### Compiling C to wasm
+
+Now that we understand how to dynamically load and execute shared objects,
+let's talk about wasm.
+In wasm we are going to generate a shared object from `sum.c`, but one that
+rather than being destined to be read by C it is for wasm.
+
+The easiest way I've seen to use `emcc` (also known as emscripten) is with
```
Comment (bzz) [inline content/post/calling-c-functions-from-bigquery.md]: Shall we also add http://emscripten.org ?

File: content/post/calling-c-functions-from-bigquery.md (hunk 3 above)
Comment (campoy) [reply content/post/calling-c-functions-from-bigquery.md]: done

File: content/post/calling-c-functions-from-bigquery.md (hunk 4 above)
Comment (campoy) [reply content/post/calling-c-functions-from-bigquery.md]: sure!

**Comment by campoy** [comment]
This is now published, but I let it up just in case there's any other corrections anyone has

Please approve the changes when you have a minute 😄

**Comment by vmarkovtsev** [comment]
Merging this - it is already published as https://blog.sourced.tech/post/calling-c-functions-from-bigquery/
BTW it is a mystery to me how it was possible, @dpordomingo ?

**Pull request merged**